### PR TITLE
Use native nav bar

### PR DESCRIPTION
**Pull request by DmitrySvetlichny**

### CODE REVIEW

File: Sources/Controllers/TargetMenu/GPX/TrackMenu/OADeleteWaypointsViewController.mm [hunk 1]
```diff
@@ -81,68 +74,58 @@ -(void)viewWillDisappear:(BOOL)animated
...
+- (NSAttributedString *)getBottomButtonTitleAttr
 {
     BOOL hasSelection = _selectedWaypointGroups.allKeys.count != 0;
-    self.deleteButton.backgroundColor =
-            hasSelection ? UIColor.buttonBgColorDisruptive : UIColor.buttonBgColorDisabled;
-    [self.deleteButton setTintColor:hasSelection ? UIColor.buttonTextColorPrimary : UIColor.textColorSecondary];
-    [self.deleteButton setUserInteractionEnabled:hasSelection];
 
-    NSString *textShow = OALocalizedString(@"shared_string_delete");
+    NSString *textShow = OALocalizedString( @"shared_string_delete");
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/GPX/TrackMenu/OADeleteWaypointsViewController.mm:101]: remove extra space

File: Sources/Controllers/RouteInfoMenu/OARouteSettingsBaseViewController.mm [hunk 2]
```diff
@@ -82,6 +76,29 @@ - (void) didReceiveMemoryWarning
 - (void) viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```
Comment (tigrim) [inline Sources/Controllers/RouteInfoMenu/OARouteSettingsBaseViewController.mm:80]: The code for styling the navigation bar should be placed in a separate method, configureNavigationBar

File: Sources/Controllers/RoutePlanning/OASaveTrackViewController.mm [hunk 3]
```diff
@@ -117,6 +117,23 @@ - (NSString *) getDisplayingFolderName:(NSString *)filePath
 - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```
Comment (tigrim) [inline Sources/Controllers/RoutePlanning/OASaveTrackViewController.mm:121]: The code for styling the navigation bar should be placed in a separate method, configureNavigationBar

File: Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m [hunk 4]
```diff
@@ -106,40 +103,36 @@ - (void) viewDidLoad
...
-    {
-        UIView *res = [[UIView alloc] init];
-        res.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
-        res.backgroundColor = UIColor.viewBgColor;
-        res.alpha = 0;
-        return res;
-    }
+    [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinateFormatsViewController.m:118]: The code for styling the navigation bar should be placed in a separate method, configureNavigationBar

File: Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinatesViewController.mm [hunk 5]
```diff
@@ -134,6 +131,26 @@ - (instancetype) initWithLat:(double)lat lon:(double)lon
 - (void)viewWillAppear:(BOOL)animated
 {
     [super viewWillAppear:animated];
+    
+    [self.navigationController setNavigationBarHidden:NO animated:YES];
```
Comment (tigrim) [inline Sources/Controllers/TargetMenu/Search/OAQuickSearchCoordinatesViewController.mm:135]: The code for styling the navigation bar should be placed in a separate method, configureNavigationBar

File: Sources/Controllers/RouteInfoMenu/OARouteSettingsBaseViewController.mm [hunk 6]
```diff
@@ -318,13 +335,8 @@ - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIVi
...
 - (void) setCancelButtonAsImage
 {
-    [self.backButton setTitle:nil forState:UIControlStateNormal];
-    [self.backButton setImage:[UIImage imageNamed:@"ic_navbar_chevron"] forState:UIControlStateNormal];
-}
-
-- (IBAction)doneButtonPressed:(id)sender
-{
-    [self doneButtonPressed];
+    UIBarButtonItem *backButton = [[UIBarButtonItem alloc] initWithImage:[UIImage templateImageNamed:@"ic_navbar_chevron"].imageFlippedForRightToLeftLayoutDirection style:UIBarButtonItemStylePlain target:self action:@selector(onLeftNavbarButtonPressed)];
```
[the remark this answers is not in the document]
Comment (DmitrySvetlichny) [reply Sources/Controllers/RouteInfoMenu/OARouteSettingsBaseViewController.mm:338]: YES

File: Sources/Controllers/BottomSheet/RouteInfoBottomSheet/OAFollowTrackBottomSheetViewController.mm [hunk 7]
```diff
@@ -470,8 +470,8 @@ - (void)presentOpenTrackViewController:(BOOL)animated
     _navigationController.navigationBarHidden = YES;
     OAOpenAddTrackViewController *saveTrackViewController = [[OAOpenAddTrackViewController alloc] initWithScreenType:EOAFollowTrack];
     saveTrackViewController.delegate = self;
-    [_navigationController setViewControllers:@[saveTrackViewController]];
```
[the remark this answers is not in the document]
Comment (DmitrySvetlichny) [reply Sources/Controllers/BottomSheet/RouteInfoBottomSheet/OAFollowTrackBottomSheetViewController.mm:473]: No!

**Review by tigrim — APPROVED** [review]

**Pull request merged**